### PR TITLE
[P2P] refactor: rename "network" to "router"

### PR DESCRIPTION
**Pull request by bryanchriswhite**
## Description

Refactors everything in the P2P module which previously referred to a "network" to instead be thought of as a "router". This begins with the `typesP2P.Network` interface and the changes propagates down from there, through implementations and into variable names, tests, etc.

## Issue

Fixes #<issue_number>

## Type of change

Please mark the relevant option(s):

- [ ] New feature, functionality or library
- [ ] Bug fix
- [x] Code health or cleanup
- [ ] Major breaking change
- [ ] Documentation
- [ ] Other 

## List of changes

- Renamed `Network` interface to `Router`
- Shortened `Router#NetworkBroadcast` to `#Broadcast`
- Shortened `Router#NetworkSend` to `#Send`
- Shortened `Router#networkSendInternal` to `#sendInternal`
- Shortened `Router#networkBroadcastAtLevel` to `#broadcastAtLevel`
- Renamed `rainTreeNetwork` to `rainTreeRouter`
- Renamed `rainTreeNetwork` method receivers
- Renamed `p2pModule#network` to `#router`
- Renamed `p2pModule#setupNetwork()` to `#setupRouter()`
- Renamed config var in `rainTreeRouter#Create`
- Renamed router logger
- Renamed `NewRainTreeNetwork()` to `NewRainTreeRouter()`
- Refactored peers_manager_test.go
- Refactored network_test.go
- Simplified p2p module/router config handoff
- Updated debug logging

## Testing

- [ ] `make develop_test`; if any code changes were made
- [ ] [Docker Compose LocalNet](https://github.com/pokt-network/pocket/blob/main/docs/development/README.md); if any major functionality was changed or introduced
- [ ] [k8s LocalNet](https://github.com/pokt-network/pocket/blob/main/build/localnet/README.md); if any infrastructure or configuration changes were made

## Required Checklist

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added, or updated, [`godoc` format comments](https://go.dev/blog/godoc) on touched members (see: [tip.golang.org/doc/comment](https://tip.golang.org/doc/comment))
- [ ] I have tested my changes using the available tooling
- [ ] I have updated the corresponding CHANGELOG

### If Applicable Checklist

- [ ] I have updated the corresponding README(s); local and/or global
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added, or updated, [mermaid.js](https://mermaid-js.github.io) diagrams in the corresponding README(s)
- [ ] I have added, or updated, documentation and [mermaid.js](https://mermaid-js.github.io) diagrams in `shared/docs/*` if I updated `shared/*`README(s)

### CODE REVIEW

File: consensus/state_sync/interfaces.go [hunk 1]
```diff
@@ -91,10 +91,10 @@ type StateSyncModuleLEGACY interface {
 	// An eligible peer is when `PeerMeta.MinHeight <= blockHeight <= PeerMeta.MaxHeight`
 	GetRandomEligiblePeersForHeight(blockHeight int64) (eligiblePeer PeerSyncMeta, err error)
 
-	// Uses `NetworkSend` to send a `BlockRequestMessage` to a specific peer
+	// Uses `typesP2P.Router#Send()` to send a `BlockRequestMessage` to a specific peer
 	SendBlockRequest(peerId string) error
 
-	// Uses 'NetworkSend' to send a `BlockResponseMessage` to a specific peer
+	// Uses 'typesP2P.Router#Send()' to send a `BlockResponseMessage` to a specific peer
```
Comment (dylanlott) [inline consensus/state_sync/interfaces.go:97]: :100:

File: p2p/types/router.go [hunk 2]
```diff
@@ -9,11 +9,11 @@ import (
 
 // TECHDEBT(olshansky): When we delete `stdnetwork` and only go with `raintree`, this interface
 // can be simplified greatly.
-type Network interface {
+type Router interface {
```
Comment (dylanlott) [inline p2p/types/router.go:12]: Very much agree that Router is better than Network. It's clearer what it's actually doing in the code.

File: p2p/types/router.go [hunk 3]
```diff
@@ -9,11 +9,11 @@ import (
...
 	modules.IntegratableModule
 
-	NetworkBroadcast(data []byte) error
-	NetworkSend(data []byte, address cryptoPocket.Address) error
```
Comment (dylanlott) [inline p2p/types/router.go:16]: You love to see it go 🙌

**Review by dylanlott — APPROVED** [review]
Passes tests on my machine, LGTM 🚀

**Review by Olshansk — APPROVED** [review]
❤️

**Pull request merged**